src/utils/Cache.ts

```diff
@@ -0,0 +1,246 @@
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+
+export type CacheTargets = CacheTarget[];
+export enum CacheTarget {
+  CACHE_API = 'CACHE_API',
+  MEMORY = 'MEMORY',
+}
+export type CacheResponse = {
+  data: any;
+  status: number;
+  statusText: string;
+  headers: any;
+};
+
+export const CACHE_API_KEY = 'sentinelhub-v1';
+export const SUPPORTED_TARGETS = [CacheTarget.CACHE_API, CacheTarget.MEMORY];
+const FALLBACK_TARGET = CacheTarget.MEMORY;
+
+export const memoryCache = new Map();
+
+// Factory will return an instance of a Cache interface.
+// The first availble target will be used
+// By default we will use cache_api if availble, if not we will cache in memory
+// user can also specify to just use in memory caching
+// If user provides a CacheTarget.CACHE_API and cache_api is not availble we will fallback to memory
+export function cacheFactory(optionalTargets: CacheTargets): ShCache {
+  const targets = optionalTargets || SUPPORTED_TARGETS;
+  const target = getFirstUsuableTarget(targets);
+  return constructCache(target);
+}
+
+function getFirstUsuableTarget(targets: CacheTargets): CacheTarget {
+  let firstTargetToUse = FALLBACK_TARGET; // default to memory if target is not supported
```

Comment:
Maybe a better idea would be to default to _no cache_ instead of memory? If developer explicitly set just `[CACHE_API]`, they would probably assume that memory would not be used as target?

Reply:
Resolving since defaulting to no cache was added.

Reply:
This has been changed so that we do not fallback to memory cache. If the user doesnt supply targets then the default targets will be used and will sill fallback to memory if the other targets are undefined